dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -493,13 +493,12 @@ CDefaultContextMenu::AddStaticContextMenusToMenu(
     UINT uFlags)
 {
     UINT ntver = RosGetProcessEffectiveVersion();
-    MENUITEMINFOW mii;
+    MENUITEMINFOW mii = { sizeof(mii) };
```

Comment:
Does this ensure that the other fields of `mii` get zero-initialized?

Reply:
Yes

Reply:
Yes, it does. But I always wondered by that kind of optimization who will set .cbSize properly then...

Reply:
That brace syntax initializes the members you specify (in struct order), the remaining members are zero initialized (a language lawyer might claim they are default initialized, I don't feel like looking it up. For POD members that is zero anyway).

Reply:
Yes, sure. But in my head it would not work like that, if .cbSize **was** the last member. Do I assume correctly? Is MS always placing the .cbSize first in all their structures? Or are there any structures where they put it in a different order and where we would need to be careful?

Reply:
The size member is always first. The braces are not magic, it would not work if it was the last member.